hpx/parallel/algorithms/detail/predicates.hpp

```diff
@@ -116,7 +155,11 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1) { namespace detail
         HPX_HOST_DEVICE
         HPX_FORCEINLINE static Iter call(Iter iter, Stride offset)
         {
+#if defined(HPX_HAVE_CUDA) && defined(__CUDACC__)
+            hpx::detail::advance(iter, offset);
+#else
             std::advance(iter, offset);
+#endif
```

Comment:
Not sure about which kind of optimizations vendor compilers are doing with `std::advance`. I suggest to let it this way to preserve the host side `advance`